### ISSUE TITLE
Use the <Card> WordPress component

### ISSUE THREAD

**Issue by dechov** [woocommerce/woocommerce-admin#4002]
As noted by @LevinMedia, all uses of the `<Card>` _WooCommerce_ component should be switched to the `<Card>` _WordPress_ component [[src](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/card), [story](https://wordpress.github.io/gutenberg/?path=/story/components-card--default)]. (cc @jameskoster)

One possible approach to consider would be a deprecation path in which the `<Card>` _WooCommerce_ component would be reimplemented as a wrapper for the WordPress component (unless there is some incompatibility I'm unaware of). This way, all downstream usage (inside and outside of this repository) would use the "common" component sooner rather than later, and refactoring could proceed independently.

(For extra context: WooCommerce Payments is currently using `<Card>` from `@woocommerce/components` both [directly](https://github.com/Automattic/woocommerce-payments/blob/ddc2ebc1ce8305e02d93f46b9f687d57d6f6b08d/client/transaction-details/summary/index.js#L9) and [indirectly](https://github.com/Automattic/woocommerce-payments/blob/ddc2ebc1ce8305e02d93f46b9f687d57d6f6b08d/client/transactions/index.js#L10).)

**Comment by jameskoster** [comment]
One potential use case for keeping the wc-admin Card component around:

We've often spoken about how the padding values on a Card should adapt depending on its rendered dimensions. The obvious example is that the padding should be smaller on phones than it is on desktop. 

So on a super-basic level, perhaps the wc-admin Card component could contain logic to switch the `size` of the core Card component from `medium` to `small` based on the browser width?

Extra bonus points if we can work out a way to implement container queries, and adjust the padding based on that!

cc @LevinMedia

**Comment by dechov** [comment]
Dynamically switching the value based on browser width is possible – at one time I was doing a similar thing across so many projects that I'd [packaged a higher-order component](https://github.com/two-n/react-sizeable) for facilitating it (except it reads parent node size, but in response to browser resize).

I would leave it to Isotope to determine if it's advisable in this context – I don't see why not, though if it can somehow be solved in CSS then it probably should be.

**Comment by jameskoster** [comment]
It looks like your `Sizeable` component is basically solving the container query problem, is that correct?

Would it enable us to - for example - set the `size` of the `Card` to `small` if the _parent element_ was narrower than 375px, and to `medium` if it was wider than 375px?

If so, that would be awesome! Afaik this is not currently possible with CSS alone.

As a fallback, a swap based on _browser width_ would also be acceptable. While this _is_ possible with CSS alone, it would preferable to swap the `size` prop on the core `Card` component instead, to ensure we always inherit any padding changes there.

**Comment by mattsherman** [comment]
I agree that we should deprecate the `Card` from `@woocommerce/components` in favor of the one from `@wordpress/components`. The wrapper approach would be a good interim step.

**Comment by dechov** [comment]
> It looks like your Sizeable component is basically solving the container query problem, is that correct?

I believe so, though I wasn't (and am not) well versed in the full nuance of the problem. I'd only done it based on browser width, but I think it could be done based on parent element width, especially if _that_ only changes in response to browser resize.

I'd made a quick attempt some weeks ago to adapt the WC Card (replacing a `div`, a `div`, and a `Section` with `Card`, `CardHeader`, and `CardBody`, respectively) and it seemed largely OK though with an assortment of styling glitches.

Not sure about any other contexts but in WCPay we use footers in a couple of instances (and have [our own `CardFooter` component](https://github.com/Automattic/woocommerce-payments/tree/master/client/components/card-footer)). It has its own CSS, and would need to be adapted to the WP Card – ideally we could use `CardFooter` from `@wordpress/components`, but it should not be inside the body as it currently is.

In that vein, the last thing I tried was detecting whether a `CardBody` was among the `children` so as to avoid wrapping it in that case (also meaning that a `CardFooter` sibling element could be present), but I think it would need to be matched by component name ("CardBody") since the different instances of `@wordpress/components` in different bundles lead to technically separate components.

I made no attempt at auto-sizing the card based on browser width, but I think it is feasible.
